### PR TITLE
fix: ButtonGroup style bug when have disabled button #9296

### PR DESCRIPTION
**Pull request by tangjinzhou**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [ ] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [ ] Update API docs for the component.
  * [ ] Update/Add demo to demonstrate new feature.
  * [ ] Update TypeScript definition for the component.
  * [ ] Add unit tests for the feature.

https://codesandbox.io/s/61lqr2m4mr
image

### CODE REVIEW

File: components/button/style/mixin.less [hunk 1]
```diff
@@ -260,25 +263,50 @@
...
-  .@{btnClassName}:not(:first-child):not(:last-child) {
+  .@{btnClassName} {
     border-radius: 0;
   }
 
   > .@{btnClassName}:first-child,
   > span:first-child > .@{btnClassName} {
     margin-left: 0;
   }
+  > .@{btnClassName}:first-child:last-child {
```
Comment (afc163) [inline components/button/style/mixin.less]: `:only:child`

**Comment by tangjinzhou** [comment]
#9296

File: components/button/style/mixin.less (hunk 1 above)
Comment (tangjinzhou) [reply components/button/style/mixin.less]: 👍 :only-child

**Comment by zombieJ** [comment]
Thanks for contribution, merge into master

**Pull request merged**